src/content_sao.cpp

```diff
@@ -443,31 +443,28 @@ int LuaEntitySAO::punch(v3f dir,
 			punchitem,
 			time_from_last_punch);
 
-	if(result.did_punch)
-	{
+	if (result.did_punch) {
 		setHP(getHP() - result.damage);
-
-
 		std::string punchername = "nil";
 
-		if ( puncher != 0 )
-			punchername = puncher->getDescription();
+		if (result.damage > 0) {
+			if (puncher != 0)
```

Comment:
`std::string punchername = puncher ? puncher->getDescription() : "nil";` and remove line 448

If you do not like the ternary operation I'd still suggest removing line 448 and having here

```
std::string punchername;
if (puncher == NULL)
    punchername = "nil";
else
   punchername = puncher->getDescription();
```

If you don't, for some bizarre reason (heh), want to do either of those changes then the line has to at least be changed from `if (puncher != 0)` to `if (puncher != NULL)`, or even better simply `if (puncher)`